src/clients/authorization/policy/Group.tsx

```diff
@@ -51,112 +51,132 @@ export const Group = () => {
   );
 
   return (
-    <FormGroup
-      label={t("groups")}
-      labelIcon={
-        <HelpItem
-          helpText="clients-help:policyGroups"
-          fieldLabelId="clients:groups"
+    <>
+      <FormGroup
+        label={t("groupsClaim")}
+        labelIcon={
+          <HelpItem
+            helpText="clients-help:groupsClaim"
+            fieldLabelId="clients:groupsClaim"
+          />
+        }
+        fieldId="groups"
+      >
+        <TextInput
+          type="text"
+          id="groupsClaim"
+          name="groupsClaim"
+          data-testid="groupsClaim"
+          ref={register}
         />
-      }
-      fieldId="groups"
-      helperTextInvalid={t("requiredGroups")}
-      validated={errors.groups ? "error" : "default"}
-      isRequired
-    >
-      <Controller
-        name="groups"
-        control={control}
-        defaultValue={[]}
-        rules={{
-          validate: (value: GroupValue[]) =>
-            value.filter((c) => c.id).length > 0,
-        }}
-        render={({ onChange, value }) => (
-          <>
-            {open && (
-              <GroupPickerDialog
-                type="selectMany"
-                text={{
-                  title: "clients:addGroupsToGroupPolicy",
-                  ok: "common:add",
+      </FormGroup>
+      <FormGroup
+        label={t("groups")}
+        labelIcon={
+          <HelpItem
+            helpText="clients-help:policyGroups"
+            fieldLabelId="clients:groups"
+          />
+        }
+        fieldId="groups"
+        helperTextInvalid={t("requiredGroups")}
+        validated={errors.groups ? "error" : "default"}
+        isRequired
+      >
+        <Controller
+          name="groups"
+          control={control}
+          defaultValue={[]}
+          rules={{
+            validate: (value: GroupValue[]) =>
+              value.filter((c) => c.id).length > 0,
+          }}
+          render={({ onChange, value }) => (
+            <>
+              {open && (
+                <GroupPickerDialog
+                  type="selectMany"
+                  text={{
+                    title: "clients:addGroupsToGroupPolicy",
+                    ok: "common:add",
+                  }}
+                  onConfirm={(groups) => {
+                    onChange([
+                      ...value,
+                      ...groups.map((group) => ({ id: group.id })),
```

Comment:
Also this can be shortened with destructuring.
```suggestion
                      ...groups.map(({ id }) => ({ id })),
```